etcdmain/config.go

```diff
@@ -127,9 +133,66 @@ type config struct {
 
 	enablePprof bool
 
+	configFile string
+
 	ignored []string
 }
 
+type yamlSecConfig struct {
+	CAFile        string `json:"ca-file"`
+	CertFile      string `json:"cert-file"`
+	KeyFile       string `json:"key-file"`
+	CertAuth      bool   `json:"client-cert-auth"`
+	TrustedCAFile string `json:"trusted-ca-file"`
+}
+
+type yamlConfig struct {
```

Comment:
`fallback`, `proxy`, `clientTLSInfo, peerTLSInfo` also cannot be converted from configuration file directly. So we have to keep them into configuration. We can convert the corresponding public one `Fallback`, `Proxy` and `XXXTLSInfo` to the private ones after we finish parsed the configuration file. Does it make sense?